internal/terminal/table.go

```diff
@@ -0,0 +1,131 @@
+package terminal
+
+import (
+	"errors"
+	"fmt"
+	"github.com/fatih/color"
+	"strings"
+)
+
+type table struct {
+	headers      []string
+	data         []map[string]string
+	columnWidths map[string]int
+}
+
+const (
+	// gutterLength is to set a standard for the break between columns for the table
+	gutterLength = 2
+)
+
+func parseInterface(data interface{}) string {
+	var normalizedData string
+	switch v := data.(type) {
+	case nil: // leave zero data
+		normalizedData = ""
+	case fmt.Stringer:
+		normalizedData = v.String()
+	default:
+		normalizedData = fmt.Sprintf("%v", v)
+	}
+	return normalizedData
+}
+
+func newTable(headers []string, data []map[string]interface{}) table {
+	resultTable := table{
+		[]string{},
+		[]map[string]string{},
+		map[string]int{},
+	}
+
+	if len(headers) == 0 {
+		return resultTable
+	}
+	resultTable.headers = headers
+
+	for _, header := range headers {
+		if headerLen := len(header); headerLen > resultTable.columnWidths[header] {
+			resultTable.columnWidths[header] = headerLen
+		}
+	}
+
+	for _, row := range data {
+		normalizedRow := make(map[string]string)
+		for key, rowData := range row {
```

Comment:
Since we need to go through all the values for each header, does it matter if we go through them in the header order? Is there a performance difference with looping through a map vs an array?

Reply:
what this ensures is our `normalizedData` will be stripped of any row-cell values that are not specified by any of the headers.

i'm thinking of a situation like:

```go
table := newTable(
  []string{"a", "b", "c"},
  []map[string]interface{}{
    "a": "first",
    "b": "second",
    "c": "third",
    "d": "will not show up in Payload() now",
  },
)
```